django/forms/widgets.py

```diff
@@ -573,6 +573,24 @@ def render_options(self, selected_choices):
                 output.append(self.render_option(selected_choices, option_value, option_label))
         return '\n'.join(output)
 
+    def choice_has_empty_value(self, choice):
+        choice_value = choice[0]
+        if isinstance(choice_value, six.string_types):
+            return len(choice_value) == 0
+        return choice_value is None
+
+    def use_required_attribute(self, initial):
+        # If first item has a value, rendering required attribute causes HTML
```

Comment:
Could be more clear about "item" refers to. This might be better as a docstring since an inline comment makes me think it's referring to the next line or block.